app/scripts/controllers/network/network-controller.test.js

```diff
@@ -2409,29 +2446,41 @@ describe('NetworkController', () => {
           });
           network.mockEssentialRpcCalls();
 
-          controller.setRpcTarget('https://mock-rpc-url', '0x1337');
+          controller.setActiveNetwork('testNetworkConfigurationId');
 
-          expect(controller.store.getState().provider).toStrictEqual({
+          expect(
+            controller.store.getState().previousProviderStore,
+          ).toStrictEqual({
             type: 'rpc',
-            rpcUrl: 'https://mock-rpc-url',
-            chainId: '0x1337',
-            ticker: 'ETH',
-            nickname: '',
-            rpcPrefs: undefined,
+            rpcUrl: 'http://example-custom-rpc.metamask.io',
+            chainId: '0x9999',
+            chainName: 'Test initial state',
           });
         },
       );
     });
 
-    it('overwrites the provider configuration completely given a maximal set of arguments', async () => {
+    it('overwrites the provider configuration given a networkConfigurationId that matches a configured networkConfiguration', async () => {
       await withController(
         {
           state: {
             provider: {
               type: 'rpc',
               rpcUrl: 'http://example-custom-rpc.metamask.io',
```

Comment:
Nit: Maybe we should initialize the controller to some valid state in these tests. Here it looks like the current controller state is invalid because it's set to a network with type `rpc` that doesn't exist in the configuration.

We could update this to be initialized with a built-in network, so that the test would remain mostly unchanged but in a valid state at all times.

Reply:
Yes that makes sense. I'll run through them and make sure they are initialized with valid state configurations.

Reply:
done [here](https://github.com/MetaMask/metamask-extension/pull/17421/commits/13c9a81e49cb543166aa9a22fc9a71a94b270420)